### PR TITLE
Update token metadata HIPs with proposed mirror node REST API responses.

### PR DESCRIPTION
**Pull request by jascks**
**Description**:
This PR adds Hedera REST (mirror node) API response updates to reflect the token `metadata` and `metadata_key` fields defined in HIP-646, HIP-657 and HIP-765.

### CODE REVIEW

File: HIP/hip-646.md [hunk 1]
```diff
@@ -94,6 +94,102 @@ message TokenInfo {
...
+            "admin_key": {
+                "_type": "ED25519",
+                "key": "0aa8e21064c61eab86e2a9c164565b4e7a9a4146106e0a6cd03a8c395a110e92"
+            },
+            "metadata": "VEVTVF9tZXRhZGF0YQ==",
+            "symbol": "non_fungible",
+            "token_id": "0.0.1036",
+            "type": "NON_FUNGIBLE_UNIQUE"
+        },
+        ...
```
Comment (steven-sheehy) [inline HIP/hip-646.md:129]: Doesn't render

```suggestion
        }
```

File: HIP/hip-646.md [hunk 2]
```diff
@@ -94,6 +94,102 @@ message TokenInfo {
...
+### Hedera REST API Changes
+
+Only for individual NFTs does the Mirror Node REST API currently return a base64 encoded representation of the `metadata` field. This will be extended to an entire token type, both fungible (this HIP) and non-fungible token collections ([HIP-765](https://hips.hedera.com/hip/hip-765)).
+
+#### /api/v1/tokens
+
+```json
+{
+    "links": {
+        "next": "/api/v1/tokens?token.id=gt:0.0.1107"
```
Comment (steven-sheehy) [inline HIP/hip-646.md:106]: It's not clear if the next link is accurate without every item, so better to just null it.
```suggestion
        "next": null
```

File: HIP/hip-646.md [hunk 3]
```diff
@@ -94,6 +94,102 @@ message TokenInfo {
 Pricing considerations:
 The price for this metadata will be identical to the cost for storage of the memo.
 
+### Hedera REST API Changes
+
+Only for individual NFTs does the Mirror Node REST API currently return a base64 encoded representation of the `metadata` field. This will be extended to an entire token type, both fungible (this HIP) and non-fungible token collections ([HIP-765](https://hips.hedera.com/hip/hip-765)).
+
+#### /api/v1/tokens
+
```
Comment (steven-sheehy) [inline HIP/hip-646.md:102]: Should clearly state that we're adding a new base64 encoded `metadata` field to the list tokens REST API. The text above is more generic and not necessarily talking about this particular API.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Requested some changes for one file. Similar changes should be made for the other files, if applicable.

File: HIP/hip-646.md [hunk 4]
```diff
@@ -94,6 +94,102 @@ message TokenInfo {
 Pricing considerations:
 The price for this metadata will be identical to the cost for storage of the memo.
 
+### Hedera REST API Changes
```
[the remark this answers is not in the document]
Comment (jascks) [reply HIP/hip-646.md]: Done

File: HIP/hip-646.md (hunk 1 above)
Comment (jascks) [reply HIP/hip-646.md]: Done

File: HIP/hip-646.md (hunk 2 above)
Comment (jascks) [reply HIP/hip-646.md]: Done

File: HIP/hip-646.md (hunk 3 above)
Comment (jascks) [reply HIP/hip-646.md:102]: Done

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**